### PR TITLE
#145 Add possibility for user to view random movie detail

### LINKED ISSUE #145: User - random movie
As a user I can select a random movie (random movie detail Will be displayed).

### PR DESCRIPTION
**Pull request by daweedek2**
After user clicks on the button to view random movie, the detail of the random existing movie will be displayed.

Closes #145

### CODE REVIEW

File: src/main/java/kostka/moviecatalog/controller/MovieDetailController.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
     public static final String IS_USER_ADULT_ATTR = "isUserAdult";
     public static final String STATUS_ATTR = "status";
     private static final String MOVIE_DETAIL_VIEW = "detail";
-    private static final String REDIRECT_MOVIE_DETAIL_VIEW = "redirect:/movies/detail/";
+    public static final String REDIRECT_MOVIE_DETAIL_VIEW = "redirect:/movies/detail/";
```
Comment (mjanousek) [inline src/main/java/kostka/moviecatalog/controller/MovieDetailController.java:28]: OPTIONAL: Wrong order of properties. Maybe you can add the check to the checkstyle if it is not there: https://checkstyle.sourceforge.io/config_coding.html#DeclarationOrder ;)

File: src/main/java/kostka/moviecatalog/controller/UserController.java [hunk 2]
```diff
@@ -58,4 +65,16 @@ public String getAllMoviesPage(final Model model) {
...
         return "allMovies";
     }
+
+    @GetMapping("randomMovie")
+    public String getRandomMovieDetail(final Model model) {
+        LOGGER.info("getting random movie detail page");
+        try {
+           Long randomMovieId = randomMovieService.getRandomMovieIdFromAllMovies();
+           return REDIRECT_MOVIE_DETAIL_VIEW + randomMovieId;
+        } catch (NoMoviesInDbException e) {
```
Comment (mjanousek) [inline src/main/java/kostka/moviecatalog/controller/UserController.java:75]: OPTIONAL: Maybe you can introduce an exception handler to redirect to all movies page in all cases where the exception is thrown.

Comment (daweedek2) [reply src/main/java/kostka/moviecatalog/controller/UserController.java:75]: Will do it in separate story :-)

File: src/main/java/kostka/moviecatalog/controller/MovieDetailController.java (hunk 1 above)
Comment (daweedek2) [reply src/main/java/kostka/moviecatalog/controller/MovieDetailController.java:28]: will be sorted out in separate MR, thanks for tip!

**Pull request merged**